src/server/server_family.cc

```diff
@@ -867,7 +867,29 @@ void ServerFamily::Info(CmdArgList args, ConnectionContext* cntx) {
 }
 
 void ServerFamily::Hello(CmdArgList args, ConnectionContext* cntx) {
-  return (*cntx)->SendOk();
+  if (args.size() > 1) {
+    string_view proto_version = ArgS(args, 1);
+
+    if (proto_version != "2") {
+      (*cntx)->SendError("NOPROTO unsupported protocol version");
+      return;
+    }
+  }
+
+  (*cntx)->StartArray(12);
+  (*cntx)->SendBulkString("server");
+  (*cntx)->SendBulkString("redis");
+  (*cntx)->SendBulkString("version");
+  (*cntx)->SendBulkString(StrCat("df-", kGitTag));
```

Comment:
`StrCat("df-", kGitTag)` is duplicated.
Can you please introduce `string GetVersion()` in `common.h` that does it?

Reply:
Also, can you please add a simple test to `dragonfly_test.cc` that covers this functionality. It should be 3-4 lines, no more.

Reply:
Ah thanks for the pointers! Will do both shortly and ping you once it's ready for re-review.

Reply:
Addressed in a few follow up commits.